src/runtime_shared.js

```diff
@@ -83,6 +83,19 @@ function updateMemoryViews() {
 #endif
 }
 
+#if SUPPORT_BIG_ENDIAN
+var nativeByteOrder32 = (() => {
+    var h16 = new Int16Array(1);
+    var h8 = new Int8Array(h16.buffer);
+    h16[0] = 42;
+    return h8[0] === 42 && h8[1] === 0; // little endian ordering
```

Comment:
Surely we must already have helpers that perform this kind of thing in emscripten?  Since all the non-atomic read/writes already work?

Reply:
Other accesses use the DataView methods.

Reply:
Similar test is performed in the runtime_debug.js, but it is only included with ASSERTIONS option.